File: Assets/ShaderAccess.js
#pragma strict
var scriptname : GameObject; 
private var shader1:Shader;
private var shader2:Shader;
var rend:Renderer;
public var mr: MeshRenderer;

function Start () {
		rend = mr;
		shader1 = Shader.Find ("Diffuse");
		shader2 = Shader.Find ("Reflective/VertexLit");
		scriptname = GameObject.Find("SpinSphere");
}

function Update () {
	if (scriptname.GetComponent(Enemy1).isMoving == false){
			rend.material.shader = shader2;
		} else {
			rend.material.shader = shader1;

		}
}

File: Assets/ShaderAccess.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ShaderAccess : MonoBehaviour {
	Material m_Material;
	public Shader shader1;
	public Shader shader2;
	public Renderer rend;
	//enemy enemy1 = new enemy ();
	// Use this for initialization
	void Start () {
		rend = GetComponent<MeshRenderer> ();
		shader1 = Shader.Find ("Diffuse");
		shader2 = Shader.Find ("Specular");


	}
	
	// Update is called once per frame
	void Update () {
		if (enemy.isMoving == false) {
			rend.material.shader = shader2;
		} else {
			rend.material.shader = shader1;

		}
	}
}
